.github/workflows/buildpack-deps-emscripten.yml

```diff
@@ -0,0 +1,34 @@
+name: buildpack-deps:emscripten
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/buildpack-deps-emscripten.yml'
+      - 'scripts/docker/buildpack-deps/Dockerfile.emscripten'
+
+jobs:
+  build:
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      DOCKER_REPOSITORY: docker.pkg.github.com/${{ github.repository }}
+      IMAGE_NAME: buildpack-deps
+      IMAGE_VARIANT: emscripten
```

Comment:
All these .yml files are almost identical. Maybe it would be possible to put it all in one file by using [job.strategy](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategy) to define a matrix of image/variant combinations?

Reply:
The problem is that they all react to different https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths triggers - and a stragety matrix would have to share the triggers, right? If not it'd be nice, but I'm not sure there's a good way to unify them even further (short of making them be an actual action this workflow calls or sth like that, but that'd be overkill).

Reply:
We could add an extra step to the job to do nothing if the diff of corresponding file is empty. This might be safer anyway because the docs say that you can't 100% rely on these triggers (e.g. [if there are more than 1000 commits or if the check times out, github runs the action anyway](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#git-diff-comparisons)).

Reply:
I liked that `job.strategy.matrix` mechanism. It will always trigger the execution of `scripts/ci/docker_upgrade.sh` for all defined strategies (the image variants). But as of now only the workflow for the image variants will be continue, where the corresponding `Dockerfile` was changed.

Reply:
I'm actually not sure this is really nice... it will have weird failing test runs for the images that didn't change, I think that'll be quite confusing... not making it an error case for the jobs and just stopping them, won't be nice either, since it'll hide the fact that an actually desired job failed.

So I'd much rather avoid all that by having the jobs separated. It's good to avoid duplication - and we did so a lot already compared to the original actions, but this duplication does have its merits...

Reply:
They don't have to appear as failed. You can skip subsequent steps with an `if`.

But anyway, the current solution isn't that bad, at least as long as the jobs are short, so keeping it is also fine in my opinion.

Reply:
Sure, but succeeding runs for images that weren't touched, might be better, but are also somewhat confusing...
It'd be best, if we had access to the file name and could use it in the ``on.pull_request.paths`` part - or some other means of unifying the code while still having separate triggers, but I don't think there is a nice way to do that just yet...

But yeah, I'd also be fine with that - then we'd actually need to check if the respective Dockerfile changed at all and we could just only call the actual ``docker_upgrade.sh`` script, if it did, and otherwise print some explanatory message (like ``<Dockerfile.*> wasn't modified, skipping.``)... (the next step already has another ``if`` that will prevent it from running in that case).

So yeah, we should avoid having failing runs all the time - which of the two options we use to achieve that I'd leave to @aarlt then :-).

Reply:
Actually: looking forward: at some point in the future we had the plan of having these actions autogenerate PRs for changing the circleci config after pushing the image... that'll make the workflow scripts more complex, so at least then, unifying them might be preferable...